Improve protected branch setting page

HesterG opened the pull request:
Main changes:

1. Change html structure of protected branch page, use [`grouped fields`](https://fomantic-ui.com/collections/form.html#grouped-fields) instead of `fields` for better margin, and wrap `grouped fields` around related `field`s, remove unnecessary `<div id="protection_box" class="fields">` outer div

2. Changed some order of field to make them more categorized, used `ui dividing header` for categorization and fine tune css.

Before:

<img width="1907" alt="Screen Shot 2023-04-27 at 14 56 19" src="https://user-images.githubusercontent.com/17645053/234783731-bce8a7ce-dfc9-4d47-a3a8-b962ebea9467.png">
<img width="1849" alt="Screen Shot 2023-04-27 at 14 56 30" src="https://user-images.githubusercontent.com/17645053/234783740-c47d314e-5e2d-4854-98fd-c88f85ef3584.png">
<img width="1872" alt="Screen Shot 2023-04-27 at 14 56 36" src="https://user-images.githubusercontent.com/17645053/234783745-18e35a75-07e8-451d-b001-f9bcf16fcab5.png">

After:

https://user-images.githubusercontent.com/17645053/235114568-da010aad-7654-4410-ab8c-5d0fce7edadb.mov



3. Changed "Enable Merge Whitelist" to radio checkbox, and added "Enable Merge" radio checkbox, which are exclusive

Before:

<img width="926" alt="Screen Shot 2023-04-28 at 13 08 29" src="https://user-images.githubusercontent.com/17645053/235059233-75790f7a-e5ea-4e1c-82c6-509fef8b84b3.png">

After:

<img width="942" alt="Screen Shot 2023-04-28 at 13 09 28" src="https://user-images.githubusercontent.com/17645053/235059367-852d1f61-8407-4126-8c79-315b9c1ffada.png">


4. Add a link to set default branch on branch list page (with reference to github)

https://user-images.githubusercontent.com/17645053/234787404-61c1c7b6-aabf-429f-a109-5b690e4e0b5a.mov

5. Removed dead codes.

wxiaoguang commented on templates/repo/settings/protected_branch.tmpl:
```diff
@@ -5,38 +5,48 @@
-
-				<div class="ui divider"></div>
+				<div class="field">
+					<label for="protected_file_patterns">{{.locale.Tr "repo.settings.protect_protected_file_patterns"}}</label>
+					<input name="protected_file_patterns" id="protected_file_patterns" type="text" value="{{.Rule.ProtectedFilePatterns}}">
```
No JS, no CSS, so these `for`/`id` could be removed now.

HesterG replied:
removed in https://github.com/go-gitea/gitea/pull/24379/commits/db24ca8f8737d0cf2d7ace3b7f5d446a068f0bae

HesterG commented on templates/repo/settings/protected_branch.tmpl:
```diff
@@ -5,38 +5,48 @@
 				{{.locale.Tr "repo.settings.branch_protection" (.Rule.RuleName|Escape) | Str2html}}
 			</h4>
 			<div class="ui attached segment branch-protection">
+				<h5 class="ui dividing header">{{.locale.Tr "repo.settings.protect_patterns"}}</h5>
```
Yes, this is a kind of [variation for header](https://fomantic-ui.com/elements/header.html#dividing), looks like `.gt-border-secondary-bottom` has similar effect, I am ok with both.

wxiaoguang replied:
My preference is: if the framework doesn't tell me to do so, and it looks counterintuitive (why a "divider" could have "content"?), then I won't do that. 

Otherwise something would break in the future.

HesterG replied:
Actually this is not a divider but a header, and it is already used in many places, like on new project page, with style defined in `base.css`

<img width="825" alt="Screen Shot 2023-04-27 at 16 14 25" src="https://user-images.githubusercontent.com/17645053/234801696-fcfd1c93-124b-4901-af61-b5d0bb69f08e.png">

<img width="412" alt="Screen Shot 2023-04-27 at 16 16 22" src="https://user-images.githubusercontent.com/17645053/234802557-b6fcbc64-3ba7-4423-8620-e4f7cf87b88b.png">

I can change this to `.gt-border-secondary-bottom` anyways as I have no preference.

wxiaoguang replied:
> Actually this is not a divider but a header, and it is already used in many places,

Oh I see, it's "dividing" not "divider", so L-G-T-M.

silverwind commented:
Too much whitespace above:

<img width="471" alt="image" src="https://user-images.githubusercontent.com/115237/234808852-71a46f1b-5c07-461f-9f48-b8d5604bf28f.png">

Too much whitespace above:

<img width="477" alt="image" src="https://user-images.githubusercontent.com/115237/234809118-651fb814-a7d1-48b7-abd4-d57287dda909.png">


Too little whitespace above, too much below:

<img width="486" alt="image" src="https://user-images.githubusercontent.com/115237/234809028-4e1708a5-23f1-4b02-9f0a-832d55205a4e.png">

HesterG commented:
> Too much whitespace above:
> 
> <img alt="image" width="471" src="https://user-images.githubusercontent.com/115237/234808852-71a46f1b-5c07-461f-9f48-b8d5604bf28f.png">
> 
> Too much whitespace above:
> 
> <img alt="image" width="477" src="https://user-images.githubusercontent.com/115237/234809118-651fb814-a7d1-48b7-abd4-d57287dda909.png">
> 
> Too little whitespace above, too much below:
> 
> <img alt="image" width="486" src="https://user-images.githubusercontent.com/115237/234809028-4e1708a5-23f1-4b02-9f0a-832d55205a4e.png">

Is this from my branch? It looks like main branch

silverwind commented:
> Is this from my branch? It looks like main branch before this PR.

Ah, I erronously had gone by the "before" screenshots. The "after" ones look fine.

silverwind commented on templates/repo/branch/list.tmpl:
```diff
@@ -7,6 +7,11 @@
 				{{.locale.Tr "repo.default_branch"}}
+				{{if and $.IsWriter (not $.Repository.IsArchived) (not .IsDeleted)}}
+				<a role="button" class="right" href="{{.RepoLink}}/settings/branches" data-tooltip-content="{{.locale.Tr "repo.settings.branches.switch_default_branch"}}">
+					{{svg "octicon-arrow-switch" 16 "gt-h-100"}}
```
If this is for vertical alignment, do instead `gt-df gt-ac` on the parent node.

HesterG replied:
Added in https://github.com/go-gitea/gitea/pull/24379/commits/eadbaf467a51eeae80c3dff59186dffdc5607d62, also added `gt-sb` for justify-content

HesterG commented:
Updates:
1. Changed some orders of fields, and further split "Pull Request" into "Pull Request Approvals" and "Pull Request Merge"
2. Changed "Enable Merge Whitelist" to radio checkbox, and added "Enable Merge" radio checkbox, which are exclusive.
 
 Also updated the description. This PR is ready for review again.

wxiaoguang commented:
Finished more fine tunes, and remove dead code

HesterG commented:
Moved "status check" into "Pull Requst Approvals" part, and updated descriptions again.

wxiaoguang commented:
I do not understand why we need a table.

But these parts don't look good to me

![image](https://user-images.githubusercontent.com/2114189/235115717-c25cf4fe-324e-4264-ba9b-7f03c7c4a594.png)

wxiaoguang reviewed (changes requested):
Need more fine tunes

silverwind commented on options/locale/locale_en-US.ini:
```diff
@@ -2096,6 +2097,8 @@ settings.event_pull_request_review = Pull Request Reviewed
 settings.event_pull_request_sync = Pull Request Synchronized
 settings.event_pull_request_sync_desc = Pull request synchronized.
+settings.event_pull_request_approvals = Pull Request Approvals
+settings.event_pull_request_merge = Pull Request Merge
```
`Merging`?

silverwind replied:
Or better rename section above to `Approve` so the both match grammatically.

wxiaoguang replied:
"merge" could be a noun. So I think it's fine?

silverwind replied:
Right

silverwind commented on web_src/css/helpers.css:
```diff
@@ -60,6 +60,8 @@ Gitea's private styles use `g-` prefix.
 }
 
+.gt-max-width-24rem { max-width: 24rem !important; }
+
```
This does not seem a general useful helper, maybe add a unique class and add it there instead?

wxiaoguang replied:
I think it will be useful in the future. I can see many hard-coded width/max-width.

`a unique class` it still needs one CSS name and at least one line.

silverwind replied:
Ok, keep it.

silverwind commented on web_src/css/repository.css:
```diff
@@ -2012,31 +2012,17 @@
 
-.repository.settings.branches .branch-protection .whitelist {
-  margin-left: 26px;
+.repository.settings.branches .branch-protection .whitelist, #statuscheck_contexts_box {
```
```suggestion
.repository.settings.branches .branch-protection .whitelist,
#statuscheck_contexts_box {
```

wxiaoguang replied:
I do not think we should add `#statuscheck_contexts_box` here.

They should share common styles.

HesterG replied:
This is for the status list to be aligned with the checkbox above, but I also think table is not a good idea, what about reverting back to the commit before table and add some padding to this #statuscheck_contexts_box div, but I am not sure what does the “share common style” mean.

wxiaoguang replied:
I will try to do some changes for the table

wxiaoguang replied:
Done in aafbb0a7639f90a30779e0a6bfbc97108b93b3a6

More unnecessary styles are removed.

HesterG replied:
Thanks.

silverwind commented on web_src/js/features/repo-settings.js:
```diff
@@ -73,20 +72,13 @@ export function initRepoSettingGitHook() {
-  }
+  if (!$('.repository.settings.branches').length) return;
+  $('.toggle-target-enabled').on('change', function () {
+    const $t = $($(this).attr('data-target'));
+    $t.toggleClass('disabled', !this.checked);
+  });
+  $('.toggle-target-disabled').on('change', function () {
+    const $t = $($(this).attr('data-target'));
+    if (this.checked) $t.addClass('disabled'); // only disable, do not auto enable
```
```suggestion
    const $target = $($(this).attr('data-target'));
    $target.toggleClass('disabled', !this.checked);
  });
  $('.toggle-target-disabled').on('change', function () {
    const $target = $($(this).attr('data-target'));
    if (this.checked) $target.addClass('disabled'); // only disable, do not auto enable
```

wxiaoguang reviewed (approved):
Remove my "Request changes". If the new commit looks good, feel free to merge.

silverwind commented on options/locale/locale_en-US.ini:
```diff
@@ -2151,21 +2154,23 @@ settings.protected_branch.delete_rule = Delete Rule
 settings.protect_whitelist_teams = Whitelisted teams for pushing:
 settings.protect_whitelist_search_teams = Search teams…
-settings.protect_merge_whitelist_committers = Enable Merge Whitelist
+settings.protect_merge_whitelist_committers = Whitelist Restricted Merge
```
Hmm the old translation sounded a bit more clear to me. "Whitelist Restricted Merge" doesn't roll of the tongue so well and might even confuse translators.

wxiaoguang replied:
```suggestion
settings.protect_merge_whitelist_committers = Enable Merge Whitelist
```
Then revert
